### DISCUSSION TITLE
Accessing Shared memory address and Raw Pointer

### DISCUSSION THREAD

**Discussion by givesun** [eclipse-iceoryx/iceoryx#1946] [Q&A]
Hello, I have few questions about the way accecing shared memory and Raw pointer.
Seeing the following page, In the first picture It describe memory address of shared memory.
https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/shared-memory-communication.md
Application 1's shared memory address is '0x40000000' and Application 2's shared memory adress is '0x30000000'.
I'm little bit confuse of this picture.
When we use shared memory to communicate between processes, The processes use same address to share data.
But Seeing the picture the address is not same, Is it explaning another situation?
Or Is this related with the following page?
https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/design/relocatable_pointer.md

**Comment by elfenpiff** [comment]
@givesun We are really happy about your interest in iceoryx and would like to answer your questions. The problem is that the answer is going extremely deep into the design of iceoryx and would maybe even require some days to grasp completely with all the side constraints. Nevertheless, I would like to provide you the answers you seek. A training over multiple days would be the right environment to go into such details.

Would you be interested to hire us for a iceoryx design deep dive training? It would help us to finance new features within this project. Also we could start a conversation if you are interested in particular features which we may can implement for you.
